extensions-contrib/sqlserver-metadata-storage/src/main/java/io/druid/metadata/storage/sqlserver/SQLServerConnector.java

```diff
@@ -0,0 +1,183 @@
+package io.druid.metadata.storage.sqlserver;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.skife.jdbi.v2.Binding;
+import org.skife.jdbi.v2.ColonPrefixNamedParamStatementRewriter;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.StatementContext;
+import org.skife.jdbi.v2.tweak.HandleCallback;
+import org.skife.jdbi.v2.tweak.RewrittenStatement;
+import org.skife.jdbi.v2.tweak.StatementRewriter;
+import org.skife.jdbi.v2.util.StringMapper;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Inject;
+import com.metamx.common.logger.Logger;
+
+import io.druid.metadata.MetadataStorageConnectorConfig;
+import io.druid.metadata.MetadataStorageTablesConfig;
+import io.druid.metadata.SQLMetadataConnector;
+
+@SuppressWarnings( "nls" )
+public class SQLServerConnector extends SQLMetadataConnector
+{
+  private static final Logger log = new Logger(SQLServerConnector.class);
+
+  private static final String PAYLOAD_TYPE = "VARBINARY(MAX)";
+  private static final String SERIAL_TYPE = "[int] IDENTITY (1, 1)";
+
+  public static final int DEFAULT_STREAMING_RESULT_SIZE = 100;
+
+  private final DBI dbi;
+
+
+  private final Set<String> TRANSIENT_SQL_CLASS_CODES = new HashSet<>(Arrays.asList(
+    "08", "53", "54", "57", "58",   // Resource Failures
+    "JW", "JZ", "S1",               // Transient Failures
+    "40"                            // Transaction Rollback
+  ));
+
+  @Inject
+  public SQLServerConnector(Supplier<MetadataStorageConnectorConfig> config, Supplier<MetadataStorageTablesConfig> dbTables)
+  {
+    super(config, dbTables);
+
+    final BasicDataSource datasource = getDatasource();
+    datasource.setDriverClassLoader(getClass().getClassLoader());
+    datasource.setDriverClassName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
```

Comment:
Not quite sure what you mean by supplied.  The `public SQLServerConnector(Supplier<MetadataStorageConnectorConfig> config, Supplier<MetadataStorageTablesConfig> dbTables)` signature is the same for PostgreSQLConnector and MySQLConnector classes.